lib/controller/controller.go

```diff
@@ -1106,3 +1110,31 @@ func (ctrl *ProvisionController) getStorageClassFields(name string) (string, map
 func claimToClaimKey(claim *v1.PersistentVolumeClaim) string {
 	return fmt.Sprintf("%s/%s", claim.Namespace, claim.Name)
 }
+
+func (ctrl *ProvisionController) fetchReclaimPolicy(storageClassName string) (v1.PersistentVolumeReclaimPolicy, error) {
+	classObj, found, err := ctrl.classes.GetByKey(storageClassName)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("StorageClass %q not found", storageClassName)
+	}
+
+	var reclaimPolicy v1.PersistentVolumeReclaimPolicy
+	switch class := classObj.(type) {
+	case *storage.StorageClass:
+		reclaimPolicy = *class.ReclaimPolicy
```

Comment:
this will cause a nil pointer dereference if the provisioner is running on an older kube version. We need to do an `if version` check before we call `fetchReclaimPolicy`, `else` we keep defaulting to `Delete`. See also https://github.com/kubernetes-incubator/external-storage/pull/392/files#diff-16daa5c3f02eee06add7f1af0a3fa901R370

Reply:
So for this case, the version should be v1.8.0, right?

Reply:
yes